packages/preview/numerical-methods/0.0.1/NumMeth.typ

```diff
@@ -0,0 +1,452 @@
+  x.push(x0)
+  while n < nMax {
+    x.push(lambda * g_x(x.last()) + (1 - lambda)*x.last())
+    n += 1
+  }
+  return x.map(n=>calc.round(n,digits:accuracy))
+  
+}
+#let NM-Iterate-FPI(g_x:calc.cos,x0:1,nMax:5,accuracy:12,return-all:false) = {
+  let result = NM-Iterate-RelaxedFPI(g_x:g_x,x0:x0,nMax:nMax,lambda:1,accuracy:accuracy,return-all:return-all)
+  return result
+}
+#let NM-Iterate-NRaphson(f_x:calc.cos,x0:1,nMax:5,accuracy:12,return-all:false) = {
+  let n = 0
+
+  if not return-all {
+    let x = x0
+    while n < nMax {
+      x = x - (f_x(x)) / (NM-Differentiate-Central(f_x:f_x,
+                                            x0:x,
+                                            h:0.0000001,
+                                            accuracy:15))
+      n += 1
+    }
+    return calc.round(x,digits:accuracy)
+  }
+  let x = ()
+  x.push(x0)
+  while n < nMax {
+    let xn = x.last()
+    x.push(xn - (f_x(xn)) / (NM-Differentiate-Central(f_x:f_x,
+                                            x0:xn,
+                                            h:0.0000000001,
+                                            accuracy:15)))
+    n += 1
+  }
+  return x.map(n=>calc.round(n,digits:accuracy))
+}
+
+#let NM-Iterate-Secant(f_x:calc.sin,x0:1,x1:0.5,nRows: 5,accuracy:5,return-all:false) = {
+  let n = 0
+  if not return-all {
+    let (a,b) = (x0,x1)
+    while n < nSteps {
+      let next = (a * f_x(b) - b * f_x(a))/(f_x(b) - f_x(a))
+      a = b
+      b = next
+      n += 1
+    }
+    return calc.round(b,digits:accuracy)
+  }
+  let xn = ()
+  xn.push(x0)
+  xn.push(x1)
+  while xn.len() < nRows {
+    let (a,b) = (xn.at(xn.len()-2),xn.last())
+    if (f_x(b) - f_x(a))==0 {
+      xn.push(0)
+    } else {
+    let next = (a * f_x(b) - b * f_x(a))/(f_x(b) - f_x(a))
+    xn.push(next)
+    }
+  }
+  return xn.map(n=>calc.round(n,digits:accuracy))
+}
+
+#let NM-Iterate-FalsePosition(f_x:calc.sin,x0:1,x1:0.5,nSteps: 5,accuracy:5,return-all:false) = {
+  if f_x(x0) * f_x(x1) > 0 {return [Error, no change of sign.]}
+  let n = 0
+  if not return-all {
+    let (a,b) = (x0,x1)
+    while n < nSteps {
+      let next = (a * f_x(b) - b * f_x(a))/(f_x(b) - f_x(a))
+      if f_x(a)*f_x(next) > 0 { //Same sign, so replace b
+        b = next
+      } else {a = next}
+      n += 1
+    }
+    return (calc.round(a, digits: accuracy), 
+            calc.round(b, digits: accuracy))
+  }
+
+  let xn = ()
+  xn.push((x0,x1))
+  while xn.len() < nSteps {
+    let (a,b) = xn.last()
+    let next = (a * f_x(b) - b * f_x(a))/(f_x(b) - f_x(a))
+    if f_x(a)*f_x(next) > 0 { //Same sign, so replace b
+        xn.push((a,next))
+      } else {xn.push((next,b))}
+    n += 1
+  }
+  return xn.map(n=>(calc.round(n.at(0), digits: accuracy),
+                        calc.round(n.at(1), digits: accuracy)))
+}
+#let NM-Iterate-Bisection(f_x:calc.sin,x0:1,x1:0.5,nSteps: 5,accuracy:5,return-all:false) = {
+  if f_x(x0) * f_x(x1) > 0 {return [Error, no change of sign.]}

```

Comment:
The [`panic`](https://typst.app/docs/reference/foundations/panic/) function may be more appropriate here.